main.tf

```diff
@@ -1,6 +1,6 @@
 locals {
   master_password      = var.create_db_instance && var.create_random_password ? random_password.master_password[0].result : var.password
-  db_subnet_group_name = var.replicate_source_db != null ? null : coalesce(var.db_subnet_group_name, module.db_subnet_group.db_subnet_group_id, var.identifier)
+  db_subnet_group_name = var.cross_region_replica != true && var.replicate_source_db != null ? null : coalesce(var.db_subnet_group_name, module.db_subnet_group.db_subnet_group_id, var.identifier)
```

Comment:
@bryantbiggs `local.db_subnet_group_name` does not seem to be used in module `db_subnet_group` on line 24. Is it correct?

Reply:
correct - we only create the subnet group when `create_db_subnet_group = true` and the name is either `db_subnet_group_name` or `identifier`. 

then, based on the logic on line 3 - we provide the subnet group name to the instance itself https://github.com/terraform-aws-modules/terraform-aws-rds/blob/e2636f94f5e7d90a0b9480bf73fe0268ffe372cf/main.tf#L89

Reply:
Yeah the issue is that if want to create a read only replica, db_subnet_group_name defaults to null, which is necessary for same region replicas but it is not correct for cross-region (different vpc, different subnets etc etc). I proposed the use of a different variable so that we can override that behaviour, and at the same time allow the use of an existing subnet group or the creation of a new one with the modukle, preserving the functionality of the module itself as is. Thank you :)

Reply:
@bryantbiggs WDYT? yes/no? :)

Reply:
I think this is ok - maybe we could change `var.cross_region_replica != true && ...` to `!var.cross_region_replica && ...`

we could also update the replica examples to add an instance in another region to ensure this works (and continues to work as intended with any future changes) https://github.com/terraform-aws-modules/terraform-aws-rds/blob/e2636f94f5e7d90a0b9480bf73fe0268ffe372cf/examples/replica-postgres/main.tf#L112

Reply:
I'm willing to do that for you if that is ok :)